content/integrations/google_cloud_bigtable/Java/_index.md

```diff
@@ -0,0 +1,424 @@
+import io.opencensus.exporter.trace.stackdriver.StackdriverTraceExporter;
+import io.opencensus.trace.config.TraceConfig;
+import io.opencensus.trace.config.TraceParams;
+import io.opencensus.trace.Tracer;
+import io.opencensus.trace.Tracing;
+import io.opencensus.trace.samplers.Samplers;
+
+import org.apache.hadoop.hbase.HColumnDescriptor;
+import org.apache.hadoop.hbase.HTableDescriptor;
+import org.apache.hadoop.hbase.TableName;
+import org.apache.hadoop.hbase.client.Admin;
+import org.apache.hadoop.hbase.client.Connection;
+import org.apache.hadoop.hbase.client.Get;
+import org.apache.hadoop.hbase.client.Put;
+import org.apache.hadoop.hbase.client.Result;
+import org.apache.hadoop.hbase.client.ResultScanner;
+import org.apache.hadoop.hbase.client.Scan;
+import org.apache.hadoop.hbase.client.Table;
+import org.apache.hadoop.hbase.util.Bytes;
+
+import java.io.IOException;
+
+public class BigtableOpenCensus implements AutoCloseable {
+    private static final String PROJECT_ID = "census-demos";
+    private static final String INSTANCE_ID = "census-demos";
+    private static final byte[] TABLE_NAME = Bytes.toBytes("Hello-Bigtable");
+    private static final byte[] COLUMN_FAMILY_NAME = Bytes.toBytes("cf1");
+    private static final byte[] COLUMN_NAME = Bytes.toBytes("greeting");
+    private static final String[] GREETINGS = {"Hello World!", "Hello Cloud Bigtable!", "Hello HBase!"};
+    private static final Tracer tracer = Tracing.getTracer();
+
+    private Admin admin;
+    private Connection connection;
+
+    public BigtableOpenCensus(String projectId, String instanceId) throws Exception {
+        // Create the admin client to use for table creation and management
+        this.connection = BigtableConfiguration.connect(projectId, instanceId);
+        this.admin = this.connection.getAdmin();
+
+        // Enable observability with OpenCensus
+        this.enableOpenCensusObservability();
+    }
+
+    @Override
+    public void close() throws IOException {
+        this.connection.close();
+    }
+
+    public static void main(String ...args) {
+        // Create the Bigtable connection
+        try (BigtableOpenCensus boc = new BigtableOpenCensus(PROJECT_ID, INSTANCE_ID)) {
+            // Now for the application code
+            try (Scope ss = tracer.spanBuilder("opencensus.Bigtable.Tutorial").startScopedSpan()) {
+                HTableDescriptor descriptor = new HTableDescriptor(TableName.valueOf(TABLE_NAME));
+
+                // Create the table
+                Table table = boc.createTable(descriptor);
+
+                // Write some data to the table
+                boc.writeRows(table, GREETINGS);
+           
+                // Read the written rows
+                boc.readRows(table);
+            
+                // Finally cleanup by deleting the created table
+                boc.deleteTable(table);
+            }
+        } catch (Exception e) {
```

Comment:
You'll need to wait for >5s before exiting the main thread, otherwise no traces will be exported (export interval is 5s).

Reply:
Dope, thanks Yang! Oh that's odd that the trace blowup doesn't have annotations from the library, that means that the bigtable Java library wasn't instrumented. I'll shelve the Java section and we'll just activate it once the library is instrumented.

Reply:
Hmm, now I see that it was instrumented almost a year ago in September 2017 
 https://github.com/GoogleCloudPlatform/cloud-bigtable-client/pull/1506/

Reply:
The client library annotations still don't show up on Stackdriver after upgrading to the latest `bigtable-hbase-dataflow` :(